### PR TITLE
Add yakbak proxy servers to mock network calls

### PR DESCRIPTION
**Pull request by mattnorris**
# Pull Request

## Description

Adds the ability to use "snapshots" of network requests to speed up testing. Instead of hitting the network, the tests will read the snapshots served by the yakbak proxy servers. 

Snapshot tests *must* be run in node, like this: 

```bash
npm test -- --package @ciscospark/plugin-teams --node --snapshots
```

Fixes https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-25274

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Test Coverage

Packages can be tested like this: 

```bash
npm test -- --package @ciscospark/plugin-teams --node --snapshots
```

**However**, Teams services are still not 100%, so not all tests are passing due to network errors. 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tooling/util/proxies.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  if (proxies && proxies.length) {
+    return Promise.all(proxies.map((proxy) => stop(proxy)));
+  }
+  return Promise.resolve();
+}
+
+/**
+ * Sets the process's environment variable given the service,
+ * e.g., HYDRA_SERVICE_URL="http://localhost:3010"
+ * @param {Object} service
```
Comment (adamweeks) [inline tooling/util/proxies.js]: Should we make service a defined jsdoc type so it is easy to look up what props it should have?

File: packages/node_modules/@ciscospark/plugin-messages/test/integration/spec/messages.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ describe('plugin-messages', function () {
   this.timeout(60000);
 
   let spark;
-  before(() => testUsers.create({count: 1})
+  before('create user', () => testUsers.create({count: 1})
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/plugin-messages/test/integration/spec/messages.js]: This change may trigger the messages plugin tests to run on this PR.

File: tooling/util/proxies.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+const crypto = require('crypto');
+const http = require('http');
+const path = require('path');
+const url = require('url');
+const yakbak = require('yakbak');
+
+/**
+ * A Cisco web service that we communicate with.
+ * @typedef {Object} Service
```
Comment (mattnorris) [inline tooling/util/proxies.js]: @adamweeks does this work?

Comment (adamweeks) [reply tooling/util/proxies.js]: Awesome, yea! 👍

Comment (adamweeks) [reply tooling/util/proxies.js]: img

**Pull request merged**